homework_12.py

```diff
@@ -0,0 +1,177 @@
+"""Homework 12"""
+
+
+# 1. Банковский вклад
+class Bank:
+    """
+    class Bank for all operations
+    """
+    client_index = 0
+    client_id = None
+    name = None
+    start_balance = None
+    years = None
+    annual_interest_rate = 12 / 100
+    capitalization_number = 4
+
+    def register_client(self, client_id, name):
+        """
+         register_client
+        """
+        # Bank.client_index += 1
+        self.name = name
+        self.client_id = client_id
+
+    def open_deposit_account(self, client_id, start_balance, years):
+        """
+         open_deposit_account
+        """
+        if self.client_id == client_id:
+            self.start_balance = start_balance
+            self.years = years
+
+    def calc_interest_rate(self, client_id):
+        """
+         calculate the interest rate
+        """
+        if self.client_id == client_id:
+            capitalization_for_period = self.annual_interest_rate / self.capitalization_number
+            money_to_return = self.start_balance
+            for _ in range(self.capitalization_number):
+                money_to_return += money_to_return * capitalization_for_period
+                print('money_to_return', money_to_return)
+            # More than one statement on a single line (multiple-statements):
+            # for n in range(self.capitalization_number):money_to_return += \
+            # Unused variable 'n' (unused-variable):
+            # (Decimal(money_to_return) * Decimal(capitalization_for_period))
+            answer_is = round(money_to_return, 2)  # до 2 знаков вместо формата + переменная!
+            return answer_is
+        return 'No money - no honey'
+
+    def close_deposit(self, client_id):
+        """
+         close_deposit
+        """
+        if self.client_id == client_id:
+            self.start_balance = 0
+
+
+bank = Bank()
+bank.register_client(client_id="0000001", name="Siarhei")
+# print(bank.client_index, bank.name)
+bank.open_deposit_account(client_id="0000001", start_balance=100000, years=1)
+# print(bank.start_balance)
+ass = bank.calc_interest_rate(client_id="0000001")
+# print(ass, '===')
+bank.close_deposit(client_id="0000001")
+
+
+# 2. Библиотека aka library;
+class Book:
+    """
+    class Book
+    """
+    book_is_reserved = False
+    book_is_taken = False
```

Comment:
It will work if there is only 1 book in the library, what if they buy another book?

Reply:
That’s true, it was mentioned in the task, but if we will change requirements -> you have to review your logic